### PR TITLE
Net Block

### PR DESCRIPTION
**Pull request by etrombly**
Implementation for issue #55 
Not sure how you want documentation done for new blocks and widgets. Here's an example config for now so you can test:

[[block]]
block = "net"
device = "eno1"
interval = 5
graph = true

### CODE REVIEW

File: src/blocks/net.rs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    }
+
+    fn view(&self) -> Vec<&I3BarWidget> {
+        if self.graph {
+            return vec![&self.output_tx, &self.graph_tx, &self.output_rx, &self.graph_rx]
+        }
+        vec![&self.output_tx, &self.output_rx]
+    }
+
+    fn click(&mut self, _: &I3BarEvent) -> Result<()> {
```
Comment (svmnotn) [inline src/blocks/net.rs]: if you are not using click, you don't need to have it.

Comment (etrombly) [inline src/blocks/net.rs]: removed it

**Comment by greshake** [comment]
Sorry for the long feedback latency, really busy right now... Will review in a couple days at most, promised! Thanks for the PR!

**Comment by greshake** [comment]
Okay, first bug to be fixed: The update interval may not be an integer number of seconds! The deserializer for duration also accepts fractions, such as `0.1`. This assumption causes a division by zero in line 103 onward:
```rust
        let rx_bytes = (current_rx - self.rx_bytes) / self.update_interval.as_secs();
        let (rx_speed, rx_unit) = convert_speed(rx_bytes);
        self.rx_bytes = current_rx;

        let current_tx = read_file(&format!("{}tx_bytes", self.device_path))?
            .parse::<u64>()
            .block_error("net", "failed to parse tx_bytes")?;
        let tx_bytes = (current_tx - self.tx_bytes) / self.update_interval.as_secs();
```

**Comment by greshake** [comment]
Next, please add documentation to the README for your block. Remember to mention what units you are using (e.g. Bits or Bytes? Mbps or MB/s etc.).

**Comment by greshake** [comment]
Also, please think about using the widget state property to change colors when downloading/uploading. Maybe you need to record somehting like highest seen bandwith to actually have an absolute value with which you can measure what is 
a "high" traffic rate, depending on the connection, s.t. you dont set the download widget to critical when only downloading with 1Kbps.

**Comment by greshake** [comment]
Next bug: There is an error in your Graph widget... After running for a minute or so, I get this error:
```rust
thread 'main' panicked at 'index out of bounds: the len is 9 but the index is 9223372036854775808', src/widgets/graph.rs:62
```

**Comment by greshake** [comment]
Code review will follow when the issues above are resolved. Happy debugging! :D

**Comment by etrombly** [comment]
I haven't been able to reproduce the error on the graph widget, what block were you using it with, and what were your settings (update interval and if you had min and max manually set)? Thanks

**Comment by greshake** [comment]
It happened with this configuration (your net block is the only one that currently uses the graph widget):
```toml
[[block]]
block = "net"
device = "wlp2s0"
interval = 1
graph = true
```

**Comment by greshake** [comment]
By the way, 9223372036854775808 happens to be the max. signed 64 bit Integer, so you may be looking at an underflow issue.

**Comment by etrombly** [comment]
looks like it was probably a divide by zero error. Please try again with the current fixes.

**Comment by greshake** [comment]
Okay, so the issue didn't reoccur for me. But there is another issue- in the documentation you write that the units are in (prefix) bits per second, but my short testing suggests it is in (prefix) bytes per second... E.g. when I do a download test with 30Mbps, it displays roughly 3-4M, which is consistent with bytes/sec, e.g. 30(Megabit/s or Mbps) / 8 bit = 3.75 (Megabytes/s or MB/s). Also, please resolve the new merge conflict.. I assume it is only there because you didn't `cargo fmt` after the last changes. Good work though :+1:

**Comment by etrombly** [comment]
updated the documentation, forgot that the base units were in bytes not bits. The conflict is from the fix I made to remove the divide by zero error. You can merge the version of graph.rs from this commit.

**Comment by greshake** [comment]
Keep the nice work up! :+1:

**Pull request merged**